internal/binarylog/binarylog.go

```diff
@@ -62,17 +88,31 @@ func GetLogger() Logger {
 //
 // Each MethodLogger returned by this method is a new instance. This is to
 // generate sequence id within the call.
-func GetMethodLogger(methodName string) MethodLogger {
+func GetMethodLogger(methodName string) MethodLoggerContext {
 	if binLogger == nil {
 		return nil
 	}
-	return binLogger.GetMethodLogger(methodName)
+	return binLogger.GetMethodLoggerContext(methodName)
+}
+
+// wrappedMethodLogger implements LogWithContext by defering to underlying
+// MethodLogger while ignoring the context.
+type wrappedMethodLogger struct {
+	methodLogger MethodLogger
+}
+
+func (wml *wrappedMethodLogger) LogWithContext(_ context.Context, lec LogEntryConfig) {
+	wml.methodLogger.Log(lec)
 }
 
 func init() {
 	const envStr = "GRPC_BINARY_LOG_FILTER"
 	configStr := os.Getenv(envStr)
-	binLogger = NewLoggerFromConfigString(configStr)
+	if bl := NewLoggerFromConfigString(configStr); bl != nil {
```

Comment:
Optional but desirable: fix `NewLoggerFromConfigString` to return a context logger instead so it doesn't need to be wrapped.

Reply:
Done. Fixed this and tests.